_posts/2022-01-21-hhvm-4.145.markdown

```diff
@@ -0,0 +1,22 @@
+---
+title: "HHVM 4.145"
+layout: post
+author: atry
+category: blog
+---
+
+HHVM 4.145 is released! This release marks the end of support of 4.137;
+4.138, 4.139, 4.140, 4.143, 4.144 remain supported, as do the 4.102 and 4.128 LTS releases.
+4.141 and 4.142 were cancelled due to the holiday season.
+
+
+# Highlights
+
+- Support throw on dynamic, e.g. `function foo(dynamic $d): void { throw $d; }`
+- Add support for attributes on class constants
```

Comment:
I would like to elaborate on this list item. The [reflection hhi](https://github.com/facebook/hhvm/blob/master/hphp/hack/hhi/reflection.hhi) does not contain a type for class constants (only type constants).

Class constant attributes must inherit from the new `ClassConstantAttribute` interface. It is currently not possible to use runtime reflection on a class constant. These attributes can (only) be parsed by static analysis tools such as hhast.

Reply:
Updated. Thank you!